cmd/amazon-cloudwatch-agent/amazon-cloudwatch-agent.go

```diff
@@ -341,31 +339,28 @@ func runAgent(ctx context.Context,
 	if err != nil {
 		return err
 	}
-
 	useragent.Get().SetComponents(cfg, c)
 
-	params := getCollectorParams(factories, provider, writer)
-
-	_ = featuregate.GlobalRegistry().Set("exporter.xray.allowDot", true)
+	params := getCollectorParams(factories, providerSettings, loggerOptions)
 	cmd := otelcol.NewCommand(params)
-
-	// Noticed that args of parent process get passed here to otel collector which causes failures complaining about
-	// unrecognized args. So below change overwrites the args. Need to investigate this further as I dont think the config
-	// path below here is actually used and it still respects what was set in the settings above.
-	e := []string{"--config=" + yamlConfigPath + " --feature-gates=exporter.xray.allowDot"}
+	// *************************************************************************************************
+	// ⚠️ WARNING ⚠️
+	// Noticed that args of parent process get passed here to otel collector which causes failures
+	// complaining about unrecognized args. So below change overwrites the args.
+	// The config path below here is actually used that was set in the settings above.
+	// docs: https://github.com/open-telemetry/opentelemetry-collector/blob/main/otelcol/command.go#L63
```

Comment:
nit: Change to permalink. This line or file may be moved or renamed in other versions.